mlir/include/mlir/Dialect/Mesh/IR/MeshOps.h

```diff
@@ -114,7 +114,7 @@ inline int64_t shardDimension(int64_t dimSize, int64_t shardCount) {
     return ShapedType::kDynamic;
 
   assert(dimSize % shardCount == 0);
-  return ceilDiv(dimSize, shardCount);
+  return llvm::divideCeilSigned(dimSize, shardCount);
```

Comment:
Using divideCeil here doesn't make sense because the preceding assert ensures that the division is exact, so this is equivalent to a normal division.